python/cudf/cudf/core/dataframe.py

```diff
@@ -4265,8 +4286,64 @@ def from_records(self, data, index=None, columns=None, nan_as_null=False):
                 df[k] = Series(data[k], nan_as_null=nan_as_null)
 
         if index is not None:
-            indices = data[index]
-            return df.set_index(indices.astype(np.int64))
+            if is_scalar(index):
+                df = df.set_index(index)
+            else:
+                df._index = as_index(index)
```

Comment:
Can we just go through the `df.index` setter here instead of using the internal attribute?

Reply:
I think we'll specifically have to set internal attribute here since we have a check for previous and new_length which will fail in dataframe initialization cases.

https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/dataframe.py#L2240-#L2242

Probably this is why rest of `DataFrame` initialization code aswell is setting index to `_index`.